aws/request/request_1_8_test.go

```diff
@@ -84,3 +84,77 @@ func TestRequest_FollowPUTRedirects(t *testing.T) {
 		t.Errorf("expect %d endpoint hits, got %d", e, a)
 	}
 }
+
+func TestNewRequest_JoinEndpointWithOperationPathQuery(t *testing.T) {
+	cases := map[string]struct {
+		HTTPPath    string
+		Endpoint    *string
+		ExpectQuery string
+		ExpectPath  string
+	}{
+		"no op HTTP Path": {
+			HTTPPath:    "",
+			Endpoint:    aws.String("https://foo.bar.aws/foo?bar=Baz"),
+			ExpectPath:  "/foo",
+			ExpectQuery: "bar=Baz",
+		},
+		"no trailing slash": {
+			HTTPPath:    "/",
+			Endpoint:    aws.String("https://foo.bar.aws"),
+			ExpectPath:  "/",
+			ExpectQuery: "",
+		},
+		"set query": {
+			HTTPPath:    "/?Foo=bar",
+			Endpoint:    aws.String("https://foo.bar.aws"),
+			ExpectPath:  "/",
+			ExpectQuery: "Foo=bar",
+		},
+		"squash query": {
+			HTTPPath:    "/?Foo=bar",
+			Endpoint:    aws.String("https://foo.bar.aws/?bar=Foo"),
+			ExpectPath:  "/",
+			ExpectQuery: "Foo=bar",
+		},
+		"trailing slash": {
+			HTTPPath:    "/",
+			Endpoint:    aws.String("https://foo.bar.aws/"),
+			ExpectPath:  "/",
+			ExpectQuery: "",
+		},
+		"trailing slash set query": {
+			HTTPPath:    "/?Foo=bar",
+			Endpoint:    aws.String("https://foo.bar.aws/"),
+			ExpectPath:  "/",
+			ExpectQuery: "Foo=bar",
+		},
+		"trailing slash sqush query": {
+			HTTPPath:    "/?Foo=bar",
+			Endpoint:    aws.String("https://foo.bar.aws/?bar=Foo"),
+			ExpectPath:  "/",
+			ExpectQuery: "Foo=bar",
+		},
+	}
```

Comment:
This case is the same as the `squash query` case. 

Also nit: s/sqush/squash

Reply:
Thanks! will fix and remove duplicates.